bakery/checker.go

```diff
@@ -574,23 +463,18 @@ func (c *caveatSquasher) add0(cond string) bool {
 		// Be safe - if we can't parse the caveat, just leave it there.
 		return true
 	}
-	switch cond {
-	case checkers.CondTimeBefore:
-		et, err := time.Parse(time.RFC3339Nano, args)
-		if err != nil || et.IsZero() {
-			// Again, if it doesn't seem valid, leave it alone.
-			return true
-		}
-		if c.expiry.IsZero() || et.Before(c.expiry) {
-			c.expiry = et
-		}
-		return false
-	case checkers.CondAllow,
-		checkers.CondDeny,
-		checkers.CondDeclared:
-		return false
+	if cond != checkers.CondTimeBefore {
+		return true
+	}
+	et, err := time.Parse(time.RFC3339Nano, args)
+	if err != nil || et.IsZero() {
+		// Again, if it doesn't seem valid, leave it alone.
```

Comment:
return true if et.IsZero()?? hmmm.. i'd rather not.. it means something went wrong on our end..

Reply:
True in this case means the caveat will be added unchanged. That is, the "something wrong" will be preserved intact, which I think is the right thing to do here.